src/EnergyPlus/CurveManager.cc

```diff
@@ -358,33 +352,35 @@ namespace CurveManager {
                                                      ErrorsFound);
             ++CurveNum;
 
+            auto &thisCurve = state.dataCurveManager->PerfCurve(CurveNum);
+
             // could add checks for blank numeric fields, and use field names for errors.
-            state.dataCurveManager->PerfCurve(CurveNum).Name = Alphas(1);
-            state.dataCurveManager->PerfCurve(CurveNum).curveType = CurveType::BiQuadratic;
-            state.dataCurveManager->PerfCurve(CurveNum).ObjectType = CurrentModuleObject;
-            state.dataCurveManager->PerfCurve(CurveNum).NumDims = 2;
-            state.dataCurveManager->PerfCurve(CurveNum).InterpolationType = InterpType::EvaluateCurveToLimits;
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff1 = Numbers(1);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff2 = Numbers(2);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff3 = Numbers(3);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff4 = Numbers(4);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff5 = Numbers(5);
-            state.dataCurveManager->PerfCurve(CurveNum).Coeff6 = Numbers(6);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Min = Numbers(7);
-            state.dataCurveManager->PerfCurve(CurveNum).Var1Max = Numbers(8);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Min = Numbers(9);
-            state.dataCurveManager->PerfCurve(CurveNum).Var2Max = Numbers(10);
+            thisCurve.Name = Alphas(1);
+            thisCurve.curveType = CurveType::BiQuadratic;
+            thisCurve.ObjectType = CurrentModuleObject;
```

Comment:
This should probably be a global `enum class EPObjectType` with `std::array<std::string_view>`.

Reply:
Yep, class structure coming...

Reply:
Oh, actually I see what you are saying now, that the object names should be an enum class with string name.  This could cross over into code generation based on the current IDD to be dynamically updated, and I suspect there is a lot of overlap with the input refactor, so I won't go down that road.  But yes, I think it would be great if the use of these magic strings was reduced in the main code.